### ISSUE TITLE
[Accessibility] : Column Aria Label is not getting set in DetailList control, using the ariaLabel prop available in IColumn interface

### ISSUE THREAD

**Issue by v-sagsah** [microsoft/fluentui#13176]
**Issue :**
Narrator/NVDA is not reading the Aria Label value, i.e. set for the column header in the DetailList.

I was trying to set the custom aria label to my column headers., by setting the aria label for the column headers using the prop that I found **"ariaLabel"** i.e. their in **"IColumn interface"**.

_ariaLabel --> Optional accessibility label (aria-label) attribute that will be stamped on to the element. If none is specified, the arai-label attribute will contain the column name._

As per the description given in https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist, we can use this prop to provide a custom aria label to the column headers.

However, I found that it is used in the sample code, given in the above website.
The prop is used for the File Type column as shown below.

![image](https://user-images.githubusercontent.com/64409491/82034051-47732280-96bb-11ea-9567-c8fffa2a2a05.png)

But, when I had inspected the code, I see the aria label is set different, from the one that has been set in the code. It seems the arial-label is getting set as the name of the column name.

![image](https://user-images.githubusercontent.com/64409491/82034267-99b44380-96bb-11ea-9834-08813536f835.png)

**Comment by ecraig12345** [comment]
Thanks for reporting this! I can repro the behavior you described, but I noticed that if you stay on the column header for longer, it does eventually read the custom ariaLabel. I also checked some earlier versions and this is the way it's behaved [even back to Fabric 6](https://codepen.io/ecraig12345/pen/xxwavpw). So while I agree it's not ideal in some ways, and the documentation could be better, I don't think we should change the behavior at this point--especially for a control as complex and with as many partner dependencies as DetailsList.

**Comment by v-sagsah** [comment]
@ecraig12345 
Thank you for the fix, but the issue is still getting repro in  (Narrator + chromium edge).
I see whenever, we specify the ariaLabel prop, to a column header, the html label tag is getting added in the code, I guess, the screen reader is reading the same values.
![image](https://user-images.githubusercontent.com/64409491/82120680-94332800-97a5-11ea-97cf-af92e3d53c9d.png)

However, I validated and found, NVDA is reading it perfectly after the reading the column name.
But, Narrator is not reading it.

can you please provide the a fix / workaround for the same ?

**Comment by ecraig12345** [comment]
I didn't change the behavior, just updated the documentation to accurately state the behavior. The current behavior is by design. If you wait longer, Narrator will also read the ariaLabel.

**Comment by v-sagsah** [comment]
@ecraig12345 I tested in Narrator, and wait for 1 min _(guess this is the min waiting time)_, but it didn't read the **ariaLabel Prop** value. 

Also, I see the documentation is not yet updated to accurately state the behavior of ariaLabel Prop in IColumn interface.

![image](https://user-images.githubusercontent.com/64409491/82180112-7d074e00-98fd-11ea-919f-f259d03cd974.png)

 Please let me know, if I am missing something here ?

**Comment by ecraig12345** [comment]
The docs are updated now.

Which browser are you using with Narrator? We primarily support Edge+Narrator.

**Comment by ecraig12345** [comment]
The docs are updated now.

With the example you linked, I'm hearing the custom ariaLabel for file type read. If you're not hearing it, please provide:
1. The specific example you're using, if not the one linked above from the website
2. The specific version of Edge
3. The specific version of Windows

**Comment by v-sagsah** [comment]
@ecraig12345 
I am using the same example that I have linked & in the below version I am able to repro it
OS: Windows 10 (Version 10.0.18363 Build 18363)
Microsoft Edge Beta Version 81.0.416.72 (Official build) (64-bit)

For me the Nrrator is not reading the below
        **_ariaLabel: 'Column operations for File type, Press to sort on File type',_**
It is only reading **"File Type Button"**

![image](https://user-images.githubusercontent.com/64409491/82237878-faf64400-9953-11ea-8a98-154b5974b01c.png)

**Comment by aneeshack4** [comment]
This doesn't repro for me either now - the file type custom aria label does get read out, I don't even have to wait as you can see in the Narrator Buddy output below:
![image](https://user-images.githubusercontent.com/4161791/82345088-78549f80-99ba-11ea-8a4d-d39e4d2fa3e7.png)
I'm on Windows Version	10.0.18363 Build 18363 and Edge Chromium Version 81.0.416.77 (latest)

**Comment by v-sagsah** [comment]
@aneeshack4 I agree it is reading the "File Type Button....", but this is not the custome aria label.

I think this is the custom _ariaLabel: 'Column operations for File type, Press to sort on File type'_, that narrator need to read.

But, in the Narrator buddy output shared by you, I am not able to see **_ariaLabel: 'Column operations for File type, Press to sort on File type',_** is getting read by the narrator.

 As per the current documentation, this ariaLabel will get read after the column name.
![image](https://user-images.githubusercontent.com/64409491/82346361-fa24e700-9a13-11ea-9e29-a27812c0b083.png)

**Comment by aneeshack4** [comment]
When I Inspect, the aria-label is shows as File Type and the "Column operations...." is inside a <label> 
![image](https://user-images.githubusercontent.com/4161791/82347011-c1a5ee80-99bc-11ea-9362-8fcc37c77b87.png)

**Comment by v-sagsah** [comment]
yes @aneeshack4 but unfortunately Narrator is not reading the text inside the <label> tag.
could you please provide a possible workaround or fix?
